drivers/sensor/apds9306/apds9306.c

```diff
@@ -221,16 +224,9 @@ static int apds9306_sample_fetch(const struct device *dev, enum sensor_channel c
 		return -EFAULT;
 	}
 
-	/* Get the measurement resolution. */
-	if (i2c_reg_read_byte_dt(&config->i2c, APDS9306_REGISTER_ALS_MEAS_RATE, &buffer)) {
-		LOG_ERR("Failed reading resolution");
-		return -EFAULT;
-	}
-
 	/* Convert the resolution into a delay time and wait for the result. */
-	resolution = (buffer >> 4) & 0x07;
-	delay = apds9306_get_time_for_resolution(resolution);
-	LOG_DBG("Measurement resolution: %u", resolution);
+	delay = avago_apds9306_measurement_time[data->resolution_idx];
+	LOG_DBG("Measurement resolution: %u", data->resolution_idx);
```

Comment:
The line will be confusing. I says resolution, but its index (register values) will be printed.